### PR TITLE
turn off auto reviews from coderabbit

### LINKED ISSUE #17112: Opt new PRs out of Code Rabbit by default
### Steps to reproduce:
1. Observe Code Rabbit summaries of several recent PRs. Examples are far too numerous to list here, but some selections:
* #17104: Mentions changes to the exit dialogue that aren't included in this PR.
* #17080: mentions command line and eSpeak updates that aren't included in this PR.
* #17048: mentions update to the Seika Notetaker driver that aren't included in this PR.

### Actual behavior:
PR summaries are being generated without user prompting. These summaries suggest that unrelated (but plausible) changes are being included in PRs, which could be extremely misleading.

### Expected behavior:
* No AI summaries are generated without explicit user request.
* When AI summaries are generated, only relevant changes are included. This may be very difficult given the nature of large language models.

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
Closes #17112 

### Summary of the issue:
CodeRabbit reviews are generally unhelpful for most PRs.
An initial review when a PR is marked as ready is usually not useful, the timing for when an AI review becomes useful is better left to the contributor or human reviewers.

Instead, a manually workflow seems much more desired.

### Description of user facing changes
Disable CodeRabbit review comments happening automatically on PRs.
PR summaries will still be added, and reviews/discussion is still available from manual prompts
Update docs.

### CODE REVIEW

**Comment by codeofdusk** [comment]
> PR summaries will still be added

This doesn't address #17112 fully.

File: .coderabbit.yml [hunk 1]
```diff
@@ -11,9 +11,9 @@ reviews:
   collapse_walkthrough: false
   poem: false
   auto_review:
-    enabled: true
+    enabled: false
     auto_incremental_review: false
-    drafts: false
+    drafts: true
```
Comment (SaschaCowley) [inline .coderabbit.yml:16]: Why has this been changed?

**Comment by XLTechie** [comment]
From my reading of Bill's issue (and my own opinion), the summaries are less useful--and more misleading, than the initial code reviews that it does.

Even reading the hidden comment summaries on my own PRs, and the walkthroughs generated, their only about 2/3 correct.

I've been tempted to remove them [the summaries], but not sure if that would damage the process.

**Comment by CyrilleB79** [comment]
Here we are discussing two topics:
* Is summary useful and desirable? (described in #17112)
* Is automatic code review useful and desirable (described here)

My opinion, as @XLTechie, is that AI code review is more useful and desirable than the summary.

Regarding the summary, the question is: would it be more useful if it was more correct? And if yes, can we configure the AI (prompt / learnings) so that it makes more correct summaries? For example one point to make him learn is the following: if the PR fixes a typo or a formatting issue in the change log, do not draw a conclusion that the fixed item has been implemented / modified; AI should double check in the code if there are related changes or not.

Re auto review, @seanbudd could you come with some example where the auto review was not desirable at all? And other examples when auto-review has been triggered too early and when humans should have triggered it manually later?

**Comment by XLTechie** [comment]
> Regarding the summary, the question is: would it be more useful if it was more correct? And if yes,

My opinion is no. The PR template already causes the submitter to provide a good summary. I don't think an AI summary is more useful to anyone than that.

So unless the AI summary somehow helps the AI itself to review the code, which I don't know but doubt, I would wonder why we bother having the summaries.

File: .coderabbit.yml (hunk 1 above)
Comment (seanbudd) [reply .coderabbit.yml:16]: currently you can't request manual reviews on draft PRs, this allows that

**Comment by seanbudd** [comment]
As reviewers/maintainers we've been finding the summaries largely helpful in many cases, particularly the diagrams. I think with experience you can use AI generated text to help inform a base understanding, with the expectation that it's imperfect, before actually reviewing the human content to confirm your understanding.

We've been largely finding the CodeRabbit review comments to be mostly noise, with ~1/5 being actually useful. Often this is because the review was prompted at the wrong stage - i.e. too early in the review process. Sometimes the diff is processed unexpectedly due to merge conflicts or the target branch changing. In many PRs such as janitorial work, small fixups, cross-branch merges, the AI reviews are entirely pointless and a waste of server time.
Disabling auto-reviews allows reviewers and contributors to opt-in to AI reviews at better times, and more judiciously.

**Comment by seanbudd** [comment]
>My opinion is no. The PR template already causes the submitter to provide a good summary. I don't think an AI summary is more useful to anyone than that.

Often the template is not followed or as thoroughly filled out as desired. Often there are language or other communication barriers with contributors. An AI summary (which can be safely edited by the contributor) is fairly useful for newer devs who are still learning to communicate dev ideas. It also helps us with communication barriers by providing an alternative explanation. I guess the big issue comes with a general understanding that generative AI tools are still quite immature, and shouldn't be treated as authoritative voice or fact.

**Comment by codeofdusk** [comment]
I'd much prefer that the summary be something requested by either the PR author or a PR reviewer, in cases where they might be helpful and with an understanding that they might not be correct.

**Comment by XLTechie** [comment]
> Often there are language or other communication barriers with contributors.

A good point, I hadn't considered that.

> many PRs such as janitorial work, small fixups, cross-branch merges, the AI 
reviews are entirely pointless and a waste of server time.

I know there is an ignore directive, but can the ones with known titles (e.g. beta to master) be exempted in the config file ahead of time?

Not relevant if this PR is adopted, of course.

**Review by SaschaCowley — APPROVED** [review]

**Pull request merged**